rasa_core/training/interactive.py

```diff
@@ -708,6 +773,32 @@ def _write_nlu_to_file(export_nlu_path, evts):
             f.write(nlu_data.as_json())
 
 
+def _write_domain_to_file(domain_path, evts, endpoint):
+    # type: (Text, List[Dict[Text, Any]], EndpointConfig) -> None
+    """Write an updated domain file to the file path."""
+
+    spec = retrieve_domain(endpoint)
+    msgs = _collect_messages(evts)
+    acts = _collect_actions(evts)
+
+    found_intents = set(m.data["intent"] for m in msgs)
+    spec_intents = [list(i.keys())[0] for i in spec["intents"]]
+    for name in found_intents:
+        if name not in spec_intents:
+            spec["intents"].append({name: {"use_entities": True}})
+
+    found_entities = [e["entity"]
+                      for m in msgs
+                      for e in m.data.get("entities", [])]
+    spec["entities"] = list(set(found_entities + spec["entities"]))
+
+    found_actions = [e["name"] for e in acts]
+    spec["actions"] = list(set(found_actions + spec["actions"]))
+
+    domain = Domain.from_dict(spec)
```

Comment:
maybe makes more sense to generalize this slightly, but it onto the domain, e.g. `domain.merge(other_domain)` and then persist - I think this might come in handy anyways.